src/Avalonia.Controls/Primitives/UniformGrid.cs

```diff
@@ -84,16 +118,32 @@ protected override Size MeasureOverride(Size availableSize)
                 }
             }
 
-            return new Size(maxWidth * _columns, maxHeight * _rows);
+            var desiredWidth = maxWidth * _columns;
+            var desiredHeight = maxHeight * _rows;
+
+            if (desiredWidth > 0d)
```

Comment:
Why is spacing only applied if the desired width/height without spacing > 0? If I have a grid containing only 0-sized children, shouldn't the spacing be applied between these 0-sized children anyway?

Reply:
I think this is the logical behavior, because in case the parent assigns the minimum size the spacing only will be visible, but if you think that spacing should be applied between a 0-sized children anyway you can remove the this condition.

Reply:
I think this behavior should be tested in other XAML frameworks and copied.

Reply:
From [UniformGrid](https://github.com/CommunityToolkit/Windows/blob/82eb9a9e066c877ff2eb31768cb2ea995f1af0b7/components/Primitives/src/UniformGrid/UniformGrid.cs#L148C9-L148C115) in the CommunityToolkit it seems that the grid requests size for the spacing even if the children themselves request zero size, but I wonder if this is logical behavior to request size only for spacing, the spacing in any case will be applied in the `ArrangeOverride` method also with 0-size children (if the parent will allocate them any size), but why should require size in `MeasureOverride` only for spacing?

Reply:
Less related, but previously I tried to port WCT UniformGrid - https://github.com/AvaloniaUI/Avalonia/pull/5627

From what I remember, UniformGrid itself didn't manage spacing. Instead it was part of Grid control, which UniformGrid inherits there.

Reply:
So:

 1) It sounds like Grid needs to get this functionality first so the UniformGrid can inherit it.
 2) Adding it to Grid was briefly attempted in the past and failed -- it is likely better to wait for the open sourcing of WinUI3 code and then just port the implementation over from there as best as possible.

Reply:
Is the right thing to do now is to close it?

Reply:
@dme-compunet from my understanding, yeah I think it would be better to close this PR in favor of an issue explaining the wanted solution. Now that winui is Open sourced as of today, one may want to take a look how they implemented it and try to adopt Avalonia grid to use it. UnifromGrid should inherit these properties afaik. 

Thank you for your contribution anyway. ❤️ Hope you keep contributing.